code/obj/storage/gang_crate.dm

```diff
@@ -167,67 +172,196 @@
 			..()
 	gear_and_gimmicks
 		New()
-			var/datum/loot_generator/lootMaster =  new /datum/loot_generator(3,2)
+			initialize_loot_master(3,2)
 			lootMaster.add_random_loot(src, GANG_CRATE_GEAR, 2)
 			lootMaster.fill_remaining(src, GIMMICK)
 			..()
 
 	guns_and_gear
 		New()
-			var/datum/loot_generator/lootMaster =  new /datum/loot_generator(3,2)
+			initialize_loot_master(3,2)
 			lootMaster.add_random_loot(src, GANG_CRATE_GUN, 1)
 			lootMaster.add_random_loot(src, GANG_CRATE_AMMO, 1)
 			lootMaster.add_random_loot(src, GANG_CRATE_GEAR, 2)
 			lootMaster.fill_remaining(src, GIMMICK)
 			..()
 
+	pickup(mob/user)
+		if (open)
+			close()
+		..()
+		if (src.layer == UNDERFLOOR)
+			src.layer = OVERFLOOR
+		var/datum/gang = user.get_gang()
+		if (gang && (trapped || tracking))
+			trapped = FALSE //if one gang member gets their mitts on it, this has done its job
+			toggle_tracking(FALSE)
+			boutput(user, SPAN_ALERT("You disarm the trap in the [src]'s handle. It's now safe to carry."))
+		if (!gang && !open && trapped && owning_gang && isliving(user))
+			var/mob/living/H = user
+			idiot = user
+			trapped = FALSE
+			toggle_tracking(FALSE)
+			icon_state = "gang_dufflebag_trap"
+			cant_self_remove = TRUE
+			cant_drop = TRUE
+			var/area/area = get_area(src)
+			playsound(src.loc, 'sound/impact_sounds/Generic_Snap_1.ogg', 50, 1)
+			boutput(user, SPAN_ALERT("As you pick up \the [src.name], a series of barbs emerge from the handle, lodging in your hand!"))
+			src.owning_gang.broadcast_to_gang("The bag [src.informant] knew about has just been stolen! Looks like it was in \the [area.name]")
+			ON_COOLDOWN(src,"bleed_msg", 30 SECONDS) //set a 30 second timer to remind players to remove this
+			idiot.setStatus("gang_trap", duration = INFINITE_STATUS)
+			H.emote("scream")
+			H.bleeding = max(1,H.bleeding)
+			SPAWN(1 SECOND)
+				bleed_process()
+
+	handle_other_remove(mob/source, mob/living/carbon/human/target)
+		if (!cant_drop || !idiot || !source.get_gang())
+			return ..()
+
+		source.visible_message("[source] rips the [src] right off [target]! Ouch!","You rip the duffle bag from [target]'s hand.")
+		idiot.emote("scream")
+		playsound(idiot.loc, 'sound/impact_sounds/Generic_Snap_1.ogg', 50, 1)
+		blood_slash(idiot, 2)
+		unhook()
+		return TRUE
+
+	proc/toggle_tracking(is_tracking)
+		if (src.tracking == is_tracking)
+			return
+
+		src.tracking = is_tracking
+
+		if (src.tracking)
+			RegisterSignal(src, XSIG_MOVABLE_AREA_CHANGED, PROC_REF(alert_gang))
+		else
+			UnregisterSignal(src, XSIG_MOVABLE_AREA_CHANGED)
+
+	proc/alert_gang(datum/component/component, area/old_area, area/new_area)
+		src.owning_gang.broadcast_to_gang("The bag [src.informant] knew about is being moved! Looks like it's been moved to \the [new_area.name]")
+		toggle_tracking(FALSE)
+
+	proc/unhook()
+		if (!idiot)
+			return
+		if (istype(idiot.l_hand, /obj/item/gang_loot) && istype(idiot.r_hand, /obj/item/gang_loot)) //this is REALLY stretching it, bub.
```

Comment:
```suggestion
		if (idiot.is_in_hands(src))
```

Reply:
does not work. this is explicitly checking for two loots.